### PR TITLE
SCA SBOM report format support

### PR DESCRIPTION
**Pull request by igorlombacx**
By submitting a PR to this repository, you agree to the terms within the [Checkmarx Code of Conduct](../docs/code_of_conduct.md). Please see the [contributing guidelines](../docs/contributing.md) for how to create and submit a high-quality PR for this repo.

### Description

Support to SBOM report format. 
Example:
.\cx results show --scan-id 8d8b1581-63aa-4e61-b8e0-c6a98effe77e --report-format sbom --report-sbom-format spdxjson 

For more details:
.\cx results show --help

### References

https://checkmarx.atlassian.net/browse/AST-23496?atlOrigin=eyJpIjoiODg2NmNmYzVjMzBiNGJjNDk1YzQ2ZDYyMjNiZmUxYWEiLCJwIjoiaiJ9

### Testing

unity and integration tests 

### Checklist

- [x] I have added documentation for new/changed functionality in this PR (if applicable).
- [x] I have updated the CLI help for new/changed functionality in this PR (if applicable).
- [x] All active GitHub checks for tests, formatting, and security are passing
- [x] The correct base branch is being used

### CODE REVIEW

File: internal/wrappers/results-sbom-http.go [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+	file, err := os.Create(targetFile)
+	if err != nil {
+		return errors.Wrapf(err, "Failed to create file %s", targetFile)
+	}
+	size, err := io.Copy(file, resp.Body)
+	if err != nil {
+		return errors.Wrapf(err, "Failed to write file %s", targetFile)
+	}
+	defer file.Close()
```
Comment (diogopcx) [inline internal/wrappers/results-sbom-http.go:125]: defer close should be called right after creation if no error occurred
also, check if close returns an err and ignore it, similar to line 109

File: internal/wrappers/client.go [hunk 2]
```diff
@@ -339,6 +339,7 @@ func SendHTTPRequestWithJSONContentType(method, path string, body io.Reader, aut
 	req, err := http.NewRequest(method, fullURL, body)
 	client := GetClient(timeout)
 	setAgentName(req)
+	req.Header.Del("content-type")
```
Comment (diogopcx) [inline internal/wrappers/client.go:342]: Why is this needed?

File: internal/commands/result.go [hunk 3]
```diff
@@ -869,6 +899,53 @@ func exportJSONSummaryResults(targetFile string, results *wrappers.ResultSummary
...
+		time.Sleep(delayValueForReport * time.Millisecond)
+	}
+	if !strings.EqualFold(poolingResp.ExportStatus, completedStatus) {
+		return errors.Errorf("SBOM generating failed - Current status: %s", poolingResp.ExportStatus)
+	}
+	err = sbomWrapper.DownloadSbomReport(poolingResp.ExportID, targetFile)
+	if err != nil {
+		return errors.Wrapf(err, "%s", "Failed downloading SBOM report")
+	}
+	if err != nil {
```
Comment (diogopcx) [inline internal/commands/result.go:941]: Duplicate error handling

File: internal/commands/result.go [hunk 4]
```diff
@@ -869,6 +899,53 @@ func exportJSONSummaryResults(targetFile string, results *wrappers.ResultSummary
...
+	}
+	if formatSbomOptions != "" && formatSbomOptions != defaultSbomOption {
+		format, err := validateSbomOptions(formatSbomOptions)
+		if err != nil {
+			return err
+		}
+		payload.FileFormat = format
+	}
+
+	poolingResp := &wrappers.SbomPoolingResponse{}
```
Comment (diogopcx) [inline internal/commands/result.go:915]: pooling -> polling

File: internal/commands/result.go [hunk 5]
```diff
@@ -719,6 +734,21 @@ func createReport(
...
 	}
+	if printer.IsFormat(format, printer.FormatSbom) {
+		targetType := printer.FormatJSON
+		if strings.Contains(strings.ToLower(formatSbomOptions), printer.FormatXML) {
+			targetType = printer.FormatXML
+		}
+		summaryRpt := createTargetName(fmt.Sprintf("%s_%s", targetFile, printer.FormatSbom), targetPath, targetType)
+		convertNotAvailableNumberToZero(summary)
+
+		// if summary.ScaIssues == -1 {
```
Comment (diogopcx) [inline internal/commands/result.go:745]: remove commented code

File: internal/commands/result.go [hunk 6]
```diff
@@ -719,6 +734,21 @@ func createReport(
...
+		if !contains(summary.EnginesEnabled, scaType) {
+			err := fmt.Errorf("to generate %s report, SCA engine must be enabled on scan summary", printer.FormatSbom)
+			return err
```
Comment (diogopcx) [inline internal/commands/result.go:748]: return the fmt call directly, no need for a var

File: internal/wrappers/results-sbom-http.go [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+	path string
+}
+
+func NewResultsSbomReportsHTTPWrapper(path string) ResultsSbomWrapper {
+	return &SbomHTTPWrapper{
+		path: path,
+	}
+}
+
+func (r *SbomHTTPWrapper) GenerateSbomReport(payload *SbomReportsPayload) (*SbomReportsResponse, *WebError, error) {
```
Comment (diogopcx) [inline internal/wrappers/results-sbom-http.go:42]: we are always returning WebError as nil. does the API return a body in failure codes? if yes, decode it, otherwise remove the return

File: internal/wrappers/results-sbom-http.go [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+		}
+		return &model, nil, nil
+	case http.StatusBadRequest:
+		return nil, nil, errors.Errorf("report format unsupported for current tenant")
+	default:
+		return nil, nil, errors.Errorf("response status code %d", resp.StatusCode)
+	}
+}
+
+func (r *SbomHTTPWrapper) GetSbomReportStatus(reportID string) (*SbomPoolingResponse, *WebError, error) {
```
Comment (diogopcx) [inline internal/wrappers/results-sbom-http.go:73]: same as above

File: internal/wrappers/results-sbom-http.go [hunk 9]
```diff
@@ -0,0 +1,129 @@
...
+	if err != nil {
+		return err
+	}
+
+	defer func() {
+		_ = resp.Body.Close()
+	}()
+
+	if resp.StatusCode != http.StatusOK {
+		return errors.Errorf("response status code %d", resp.StatusCode)
```
Comment (diogopcx) [inline internal/wrappers/results-sbom-http.go:114]: should we have weberror here as well?

**Comment by diogopcx** [comment]
Did we handle the case where the API is not present / functional?

File: internal/wrappers/results-sbom-http.go (hunk 7 above)
Comment (igorlombacx) [reply internal/wrappers/results-sbom-http.go:42]: removing the WebError :) 
Thx!

**Review by pedrompflopes — APPROVED** [review]
